onedal/basic_statistics/incremental_basic_statistics.py

```diff
@@ -96,6 +97,12 @@ def partial_fit(self, X, weights=None, queue=None):
         policy = self._get_policy(queue, X)
         X, weights = _convert_to_supported(policy, X, weights)
 
+        X = _check_array(X, dtype=[np.float64, np.float32], ensure_2d=False)
```

Comment:
https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/basic_statistics/incremental_basic_statistics.py#L143 https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L2109  _check_sample_weight and the checks on X both do finite value checking.  This addition duplicates the finite value checking, ideally finite value checking should be disabled here.

Reply:
I really need to think of a way to verify finite value checking to only occur once in development.